### PR TITLE
Command to send debug emails, and documentation of email development workflow

### PR DESCRIPTION
**Pull request by jeffsmohan**
## Description
It's been a challenge working on emails in development because setting up to see the actual emails as rendered took a lot of gymnastics. To make this easier, I've written a "send-debug-email" script to quickly trigger the send of any of our 6 email types. 

Running `docker compose exec app yarn workspace server run send-debug-email` will ask you which email to send, then send a demo version of that email. 

The PR also documents the usage of Ethereal in development for capturing and viewing sent emails. 

## Screenshots / Demo Video
image

https://github.com/usdigitalresponse/usdr-gost/assets/126250/22942d01-5058-4ff7-a7f7-c0fac4683614

image

### Automated and Unit Tests
- [ ] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [ ] Provided ticket and description
- [ ] Provided screenshots/demo
- [ ] Provided testing information
- [ ] Provided adequate test coverage for all new code
- [ ] Added PR reviewers

### CODE REVIEW

**Review by TylerHendrickson — CHANGES_REQUESTED** [review]
Thanks for putting this together – it's a huge improvement! 

I have one teeny-tiny change to request (commented in the diff), as well as a proposal to consider, the latter of which doesn't necessarily need to block approval...

While I think Ethereal is a fine solution (I was pleasantly surprised to find that account-creation didn't require me to share any personal information like my own email address in order to get up and running) I can think of some reasons (supporting offline development, free service goes away, hitting quota limits, etc.) why it might be nice to offer an alternative to a third-party service. I did a bit of digging and came across [Mailpit](https://mailpit.axllent.org/) ([repo](https://github.com/axllent/mailpit)). It seems to have feature-parity with Ethereal, and a few other features, some of which I think could be pretty useful:
- No additional setup for developers (if using docker)
- Test suite / reviewer for testing compatibility with different mail clients, which we've struggled with in the past
- Desktop / mobile / tablet layouts
- UI for reviewing headers (which I think will especially come in handy once we start focusing on analytics and SES tagging)
- A well-documented API for programmatic interaction (e.g. for testing)

If we want to include this as an option, we could add it to `docker-compose.yml`:
```yaml
  mailpit:
    image: axllent/mailpit
    container_name: gost-mailpit
    restart: always
    volumes:
      - ./docker/mailpit:/data # optional, for message retention across restarts
    ports:
      - 8025:8025
      - 1025:1025
    environment:
      MP_MAX_MESSAGES: 5000 # or something reasonable
      MP_DATA_FILE: /data/mailpit.db # optional, for message retention across restarts
      MP_SMTP_AUTH_ACCEPT_ANY: 1
      MP_SMTP_AUTH_ALLOW_INSECURE: 1
```

Paired with these env vars (in `packages/server/.env`):
```env
# Email Server:
NODEMAILER_HOST=localhost
NODEMAILER_PORT=1025
NODEMAILER_SECURE=false
NODEMAILER_EMAIL="gost@grants.usdr.localhost"
NODEMAILER_EMAIL_PW=something
```

**Comment by jeffsmohan** [comment]
Here's a draft PR of the mailpit implementation for reference... once this PR is landed, I'll rebase that one on main and then ask for review: https://github.com/usdigitalresponse/usdr-gost/pull/2893

**Review by TylerHendrickson — APPROVED** [review]
LGTM! And aligned on splitting out the mailpit changes into a separate PR.

**Pull request merged**
